react-cors-spa/src/Products.jsx

```diff
@@ -0,0 +1,38 @@
+import { useState, useCallback } from 'react';
+import ProductsTable from './ProductsTable';
+import AddProduct from './AddProduct';
+import ProductDialogue from './ProductDialogue';
+
+import Skeleton from '@mui/material/Skeleton';
+import { Box } from '@mui/material';
+
+function Products() {
+    const [loading, setLoading] = useState(true);
+    const [dialogue, setDialogue] = useState(false);
+
+    const handleLoading = useCallback(() => {
+        setLoading(false);
+    }, []);
+
+    const handleDialogue = () => {
+        if(!dialogue) {
+            setDialogue(true);
+        } else {
+            setDialogue(false);
+        }
+    }
+
+    return (
+        <Box>
+            {loading ? 
+            <Box className="addProductSkeleton">
+                <Skeleton variant="rounded" width={160} height={75} />
+            </Box>
+            : <AddProduct onDialogue={handleDialogue}/>}
+            <ProductsTable onLoading={handleLoading} />
+            {dialogue && <ProductDialogue onDialogue={handleDialogue}/>}
```

Comment:
Instead of this pattern, use the <Dialog>'s `open` property. So you could pass `dialogue` (though is should be renamed to something like `isDialogOpen` 1) because it's more descriptive and 2) what we're dealing with is the UI element and not a conversation between two people lol.

Reply:
use which components open property?

Reply:
`<Dialog>`. Currently you had `open` hardcoded in but you can make it dependent on a prop.

Reply:
I was having the add product button trigger the opening of the dialog instead:

```import { Button } from "@mui/material";
import Box from '@mui/material/Box';

import "./AddProductButton.css";
import "./AddProduct.css";

function AddProduct({ onDialogOpen }) {
    return (
        <Box className="AddProductButtonContainer">
            <Button variant="outlined" size="large" onClick={onDialogOpen}>Add Product</Button>
        </Box>
    )
}

export default AddProduct;```